File: app/src/main/java/io/inabsentia/superhangman/asynctask/AsyncDownloadWords.java
package io.inabsentia.superhangman.asynctask;

import android.os.AsyncTask;
import android.util.Log;

import java.io.IOException;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashSet;
import java.util.List;

import io.inabsentia.superhangman.helper.Utils;
import io.inabsentia.superhangman.logic.GameLogic;

public class AsyncDownloadWords extends AsyncTask<String, String, List<String>> {

    private final GameLogic gameLogic = GameLogic.getInstance();
    private final Utils utils = Utils.getInstance();

    @Override
    protected List<String> doInBackground(String... strings) {
        ArrayList<String> words = new ArrayList<>();
        Log.v(getClass().getName(), "Starting AsyncTask");

        try {
            String data = gameLogic.getUrl(utils.WORD_URL);
            data = data.replaceAll("<.+?>", " ").toLowerCase().replaceAll("[^a-z]", " ");
            words.addAll(new HashSet<>(Arrays.asList(data.split(" "))));
        } catch (IOException e) {
            e.printStackTrace();
        }

        return words;
    }

    @Override
    protected void onPostExecute(List<String> strings) {
        super.onPostExecute(strings);
        gameLogic.setWords(strings);

        for (int i = 0; i < strings.size(); i++)
            Log.v(getClass().getName(), "Word [" + i + "]: " + strings.get(i));
    }

}

File: app/src/main/java/io/inabsentia/superhangman/asynctask/AsyncDownloadWords.kt
package io.inabsentia.superhangman.asynctask

import android.annotation.SuppressLint
import android.os.AsyncTask
import android.util.Log
import io.inabsentia.superhangman.logic.GameLogic
import io.inabsentia.superhangman.singleton.App
import java.util.*
import kotlin.collections.ArrayList

class AsyncDownloadWords : AsyncTask<String, String, List<String>>() {

    private val gameLogic = GameLogic.instance

    @SuppressLint("StaticFieldLeak")
    private val app = App.instance

    override fun doInBackground(vararg strings: String): List<String> {
        val words = ArrayList<String>()
        Log.v(javaClass.name, "Starting AsyncTask")

        var data = gameLogic!!.getData(app!!.WORD_URL)
        data = data.replace("<.+?>".toRegex(), " ").toLowerCase().replace("[^a-z]".toRegex(), " ")
        words.addAll(HashSet(Arrays.asList(*data.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray())))

        Log.v(javaClass.name, "Finished AsyncTask")

        return words
    }

    override fun onPostExecute(strings: List<String>) {
        super.onPostExecute(strings)
        gameLogic!!.words = strings as ArrayList<String>
    }

}